enhancements/installer/openstack-baremetal-compute.md

```diff
@@ -0,0 +1,485 @@
+* Document how to use:
+  * specify two different worker sets in `install-config.yaml`: one for the VM
+    MachineSet, and one with Ironic-backed Nova flavors for the BM MachineSet
+
+#### Case 4. Install on VMs then add BM workers (BMs on a preexisting network)
+
+**Description:**
+
+In this case, the starting point is a fully functional, VM-based cluster
+installed on the installer-provisioned network.
+
+As a day 2 operation, the OpenShift administrator adds a new MachineSet for
+adding the new BM-based Compute nodes. The MachineSet bears the `subnetName` of
+the preexisting network onto which the BMs will boot.
+
+**Prerequisites for the user:**
+
+* Connectivity must be ensured between the preexisting subnet and the cluster's
+  subnet (eg through a router).
+* Traffic must be allowed between the machine pools (eg adding the appropriate
+  Security group rules)
+
+**Expected development work:**
+
+* Allow configuring a Neutron subnet for a given worker node pool.
+* Adapt the Certificate-Signing-Request timeouts to the BM booting times.
+* Document how to use:
+  * Ensure connectivity between machine pools
+  * Create appropriate security groups
+  * Add a MachineSet specifying an Ironic-backed Nova flavor, the correct
+    `subnetName` and the Security group.
+
+### Implementation work
+
+#### Adapt the wait-for install-complete timeout
+
+_Needed for cases 1, 2, and 3_
+
+The last step of the installer is to check for the availability of the cluster
+and report a failed installation if a timeout is exceeded. However, due to the
+longer boot times of BM nodes, the installation will take longer than on VMs.
+For this reason, the timeout is likely to be hit regularly when installing on
+BMs, also for clusters that would converge to a healthy state over time.
+
+The `wait-for install-complete` timeout mechanism should be changed in a way
+that it does not imply that the installation is failed, just because it takes
+more than 40 minutes, when installing on BMs. However, the Installer is not
+aware whether the worker nodes are VMs or BMs; a conditional timeout extention
+is therefore not possible.
+
+The most straightforward solution would be to let the user set a custom timeout
+when installing, or disable the timeout altogether.
+
+#### Adapt the Certificate-Signing-Request timeouts to the BM booting times.
+
+_Needed for all cases_
+
+In order to be authorised to join the cluster, new workers are expected to
+issue a CSR within a predefined timeout. This timeout should be modified to
+accommodate BM boot times.
+
+#### Allow multiple Compute machine pools in `install-config.yaml`
+
+_Needed for case 3_
+
+Currently, `install-config.yaml` [accepts][openshift-customization] an array of
+Compute machine pools. However, entering multiple is not supported. In order to
+be able to install on one VM machine pool and one BM machine pool, the
+customization file must accept more than one Compute machine pool.
+
```

Comment:
I think we are missing details on how or why bm workers will not participate in VIPs for api or ingress

Secondly how did we make sure ingress didn't land on the bm worker or can it.

If there are no VM workers and bm workers don't get VIPs, hope it's the ingress vip balancing supposed to work?

I think they is fine missing information on this topic here